vdr/integration_test.go

```diff
@@ -0,0 +1,191 @@
+package vdr
+
+import (
+	"io/ioutil"
+	"net/url"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/nuts-foundation/go-did"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/nuts-foundation/nuts-node/core"
+	crypto "github.com/nuts-foundation/nuts-node/crypto"
+	"github.com/nuts-foundation/nuts-node/network"
+)
+
+func TestVDRIntegration_Test(t *testing.T) {
```

Comment:
This test is way too long and complicated. It feels like you're not trusting the unit tests for the intermediate results. What is the goal of this test? Test every step or make sure everything is integrated?

Reply:
It also looks like multiple things are tested which can be broken up.

Reply:
Its indeed an integrationtest. So instead of using mocks this test hits the complete stack. Its the lifetime of 2 documents. One controller of the other. It already caucht a few weird exceptions. Let me know what you think What do you mean with complicated? My goal was to make it readeable and logic to follow.

Reply:
I don't think "syntactic sugar" (the newly introduced `NutsDocUpdater`) deserves a separate integration test. Since it just operates on the data structure and the VDR funcs, it can't do anything to that VDR can't do to cause a bug in the underlying network lib imo.

Reply:
over a 100 lines. It covers at least 2 different cases that could be split up in 2 tests?

Reply:
> I don't think NutsDocUpdater deserves a separate integration test.
Its an integrationtest for the whole VDR, not just the NutsDocUpdater.